src/prefect/engine/flow_runner.py

```diff
@@ -630,66 +635,56 @@ def determine_final_state(
-                result=default_result or Result(),
-                default_result=self.flow.result,
+def _set_upstream_states_map_states(upstream_states, upstream_mapped_states):
+    # TODO: this is gross
+    for edge, upstream_state in upstream_states.items():
+        if not edge.mapped and upstream_state.is_mapped():
+            assert isinstance(upstream_state, Mapped)  # mypy assert
+            upstream_state.map_states = upstream_mapped_states.get(
+                edge, upstream_state.map_states
             )
+            upstream_state.result = [s.result for s in upstream_state.map_states]
+
+
+def run_task(
+    task: Task,
+    state: State,
+    upstream_states: Dict[Edge, State],
+    context: Dict[str, Any],
+    flow_result: Result,
+    task_runner_cls: Callable,
+    task_runner_state_handlers: Iterable[Callable],
+    is_mapped_parent: bool = False,
+) -> State:
+    """
+    Runs a specific task. This method is intended to be called by submitting it to
+    an executor.
 
-            # if this task reduces over a mapped state, make sure its children have finished
-            for edge, upstream_state in upstream_states.items():
-
-                # if the upstream state is Mapped, wait until its results are all available
-                if not edge.mapped and upstream_state.is_mapped():
-                    assert isinstance(upstream_state, Mapped)  # mypy assert
-                    upstream_state.map_states = upstream_mapped_states.get(
-                        edge, upstream_state.map_states
-                    )
-                    upstream_state.result = [
-                        s.result for s in upstream_state.map_states
-                    ]
-
-            return task_runner.run(
-                state=state,
-                upstream_states=upstream_states,
-                context=context,
-                is_mapped_parent=is_mapped_parent,
-            )
+    Args:
+        - task (Task): the task to run
+        - state (State): starting state for the Flow. Defaults to `Pending`
+        - task_runner_cls (Callable): the `TaskRunner` class to use
+        - upstream_states (Dict[Edge, State]): dictionary of upstream states
+        - context (Dict[str, Any]): a context dictionary for the task run
+        - flow_result (Result): the `Result` associated with the flow (if any)
+        - task_runner_state_handlers (Iterable[Callable]): A list of state change
+            handlers that will be provided to the task_runner, and called whenever a task changes
+            state.
+        - is_mapped_parent (bool): a boolean indicating whether this task run is the run of a parent
+            mapped task
+
+    Returns:
+        - State: `State` representing the final post-run state of the `Flow`.
+    """
+    with prefect.context(context):
+        task_runner = task_runner_cls(
+            task=task,
+            state_handlers=task_runner_state_handlers,
+            flow_result=flow_result,
```

Comment:
Seems like some of the init calls have `flow_result` as the kwarg and others have `default_result`